controller/apps_osioclient.go

```diff
@@ -0,0 +1,115 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-wit/app"
+	witclient "github.com/fabric8-services/fabric8-wit/client"
+	"github.com/fabric8-services/fabric8-wit/goasupport"
+	goaclient "github.com/goadesign/goa/client"
+	uuid "github.com/satori/go.uuid"
+)
+
+// OSIOClient contains configuration and methods for interacting with OSIO API
+type OSIOClient struct {
```

Comment:
why do you wrap the `wit client` in an `OSIOClient` struct ? I think that your code could be more straightforward without this level of abstraction (just my thought, no need to make the change if you prefer it like this, though)